jujugui/static/gui/src/app/app.js

```diff
@@ -2705,12 +2699,12 @@ YUI.add('juju-gui', function(Y) {
         // We don't always get a new username and password when switching
         // environments; only set new credentials if we've actually gotten them.
         // The GUI will automatically log in when we switch.
-        this.env.setCredentials({
+        this.user.controller = {
```

Comment:
It does, but so would the original functionality. I'd like to put something smarter in that sorts out keys, but I think that's the next phase.